src/pathfinding_service/model/token_network.py

```diff
@@ -205,38 +205,32 @@ def handle_channel_opened_event(
         participant1: Address,
         participant2: Address,
         settle_timeout: int,
-    ) -> List[ChannelView]:
-        """ Register the channel in the graph, add participents to graph if necessary.
-
-        Corresponds to the ChannelOpened event. Called by the contract event listener. """
-        views = [
-            ChannelView(
-                token_network_address=self.address,
-                channel_id=channel_identifier,
-                participant1=participant1,
-                participant2=participant2,
-                settle_timeout=settle_timeout,
-            ),
-            ChannelView(
-                token_network_address=self.address,
-                channel_id=channel_identifier,
-                participant1=participant2,
-                participant2=participant1,
-                settle_timeout=settle_timeout,
-            ),
-        ]
+    ) -> Channel:
+        """ Register the channel in the graph, add participants to graph if necessary.
+
+        Corresponds to the ChannelOpened event. Called by the contract event listener.
+        We swap participants unless participant1 < participant2."""
+
+        if participant1 > participant2:
+            (participant1, participant2) = (participant2, participant1)
+
+        channel = Channel(
+            token_network_address=self.address,
+            channel_id=channel_identifier,
+            participant1=participant1,
+            participant2=participant2,
+            settle_timeout=settle_timeout,
+        )
+        views = channel.views
 
         for cv in views:
             self.add_channel_view(cv)
 
-        return views
+        return channel
 
     def add_channel_view(self, channel_view: ChannelView) -> None:
-        # Choosing which direction to add by execution order is not very
-        # robust. We might want to change this to either
-        # * participant1 < participant2 or
-        # * same as in contract (which would require an additional attribute on ChannelView)
-        if channel_view.channel_id not in self.channel_id_to_addresses:
+        # Only add it once per channel, not once per ChannelView
+        if channel_view.participant1 < channel_view.participant2:
             self.channel_id_to_addresses[channel_view.channel_id] = (
                 channel_view.participant1,
                 channel_view.participant2,
```

Comment:
For some reason I cannot comment below:
I wonder if we should make the graph bidirectional now and store the channel instead of the views.

Reply:
I don't think that will work when we search the graph. The effect of traversing depends on the direction, so we need a directed graph AFAIK.

Reply:
Good point.